CONTRIBUTING.md

```diff
@@ -1,10 +1,40 @@
-You want to contribute? We really appreciate that.
-Please have a look at this list to facilitate your contribution:
+# Want to contribute some code? 
+
+## We’d really appreciate your support!
+
+- If the [Github issue](https://github.com/jamulussoftware/jamulus/issues) for your feature/bug fix already exists, write a message in that issue saying you want to work on it. 
+- Otherwise, please [post on the GitHub Discussions](https://github.com/jamulussoftware/jamulus/discussions) and say that you are planning to do some coding and explain why. Then we can discuss the specification. 
+- Please only start coding after we have agreed to a specification.
+
+## Jamulus project/source code general principles
+
+### Stability
+This has been, and must continue to be, the most important requirement. It is particularly important for use cases such as the Worldjam where instability during a live performance would be unacceptable. The following two principles are designed to support this principle.
+
+### [Keep it Simple and Stupid](https://en.wikipedia.org/wiki/KISS_principle) and [Do One Thing and Do It Well](https://en.wikipedia.org/wiki/Unix_philosophy#Do_One_Thing_and_Do_It_Well)
+If a feature or function can be achieved in another way by another system or method, then that is preferable to building that feature into Jamulus. 
+
+### Focus on and improve the strengths of Jamulus
+There are other solutions for real time online jamming available. But Jamulus stands alone with its client/server system, minimalistic GUI and simple functionality. 
+
+### Source code consistency
+- Respect the existing code style: Tab size=4, insert spaces.
+- Space before and after `(` and `)`, except no space between `)` and `;`, and no space before an empty `()`.
+- All bodies of `if`, `else`, `while`, `for`, etc., to be enclosed in braces `{` and `}`, usually on separate lines.
+
+Make sure (if possible) that your code compiles on Windows/Mac/Linux.
+Do not use diff/patch to send your code changes but create a Github fork of the Jamulus code and create a Pull Request when you are done.
```

Comment:
If there are conflicts with jamulussoftware/jamulus showing when you go to raise the pull request, resolve those locally:
```
git checkout master ;# checkout your local master
git pull upstream master --tags ;# refresh from jamulussoftware/jamulus (i.e. use your upstream remote name)
git push --tags --force ;# update your fork's master
git checkout - ;# back to your working branch
git rebase master ;# replay your changes onto the latest code
git push --force ;# update your working branch on your fork
```